### ISSUE TITLE
saveoverlays sync not working

### ISSUE THREAD

**Issue by bouwew** [hansrune/domoticz-contrib#1]
I have followed your recipe to try and install overlayFS on the Pi (zero W). I'm running the latest NOOBS.
The installation went fine and all seemed to work, until I decided to update some files in my /home/pi directory:
- set to rw via the rootrw-command
- copy the new files to the /home/pi/XYZ directory
- set to ro via the rootro-command (update2)
- execute sudo /etc/init.d/saveoverlays sync && sudo reboot. This gives the following result:
```

[....] Syncing overlay filesystem sync: saveoverlaysFound running mosquitto - killing 331 ...
[info] Saving fake-hwclock.
[info] Priority sync done on /var/log/fake-hwclock.data....
[info] Syncing /home....
[info] Skipping this step: /home/.overlaysync or /home_org/.overlaysync or /home_stage not available.
[info] Syncing /var....
[info] Skipping this step: /var/.overlaysync or /var_org/.overlaysync or /var_stage not available.
[info] Sync changes to disk.
[info] Remount root as read-only.
. ok

```
After the reboot the changes are not there anymore.

Also, when I do this:
- set to rw via the rootrw-command
- copy the new files to the /home/pi/XYZ directory
- set to ro via the rootro-command
- reboot

After the reboot, the changes are not there.

Am I doing something wrong? Or is there an issues with the script?

Update: I looked at your scripts and it looks for .overlaysync -files, which are not created?

**Comment by hansrune** [comment]
Hi

Sorry . My bad. I missed this step in the writeup. I have updated the 
"Prepare the overlay directories" section

Thank you for making me aware!

To correct:

- set all to rw via the rootrw-command
- mkdir /var/.overlaysync /var_org/.overlaysync /home/.overlaysync 
/home_org/.overlaysync


Then either :
     sudo /etc/init.d/saveoverlays sync && sudo reboot
... or
     rootro && sudo reboot

Those .overlaysync subdirs are added as a safeguard / signature for not 
syncing anything else than what is saveoverlay's business'



-----
Hans-Rune Bue
+47 930 41 393


Den 2018-03-27 10:50, skrev bouwew:

**Comment by bouwew** [comment]
Thank you!

Another issue, and the solution: 

Next, I tried to update my pi to the latest software.
Did the update and then issued the rootro-command: the message returned was: `mount: / is busy`.
After some searching and using this lsof command:
`
lsof +L1; lsof|sed -n '/SYSV/d; /DEL\|(path /p;' |grep -Ev '/(dev|home|tmp|var)'
`
I realized this had to do something with me running the Pixel GUI.

After some more searching I found that I could use `sudo init 3` to stop the GUI.
Did that, and then `rootro` gave me the wanted result :)
